docassemble/AssemblyLine/data/questions/al_visual.yml

```diff
@@ -348,7 +366,119 @@ back button label: |
 continue button label: |
   Start over
 continue button field: al_start_over_confirmation
+---
+id: exit and erase answers
+question: |
+  Do you want to delete your answers?
+subquestion: |
+  Your progress on this document will be completely erased. Any
+  actions that were scheduled, such as reminders from this website,
+  will also be removed.
   
+  Click "Exit and delete my answers" to delete your progress,
+  or "Back" if you want to keep working.
+back button label: |
+  Back
+continue button label: |
+  Exit and delete my answers
+continue button field: al_exit_logout_confirmation
+---
+################################### Custom error action ################################
+---
+metadata:
+  error action: al_custom_error_action
+---
+objects:
+  - MAIN_METADATA: DAEmpty()
+---
+id: custom error action
+event: al_custom_error_action
+decoration: bug
+question: |
+  Something went wrong
+subquestion: |
+  % if interview_metadata.get("main_interview_key"):
+    <%
+      MAIN_METADATA = interview_metadata[interview_metadata["main_interview_key"]]
+    %>
+  % elif len(interview_metadata) > 1:    
+    <% 
+      del(interview_metadata["main_interview_key"]) # DADict creates the key on lookup above
+      MAIN_METADATA = next(iter(interview_metadata.values())) 
+    %>
+  % else:
+    <% MAIN_METADATA = {} %>
+  % endif
+
+  We ran into an error when we tried to load this screen.
 
+  You can try one of these steps to recover your work:
 
+  * Click "Back" and try again.
+  % if al_enable_error_action_feedback_link:
+  * [:comment-dots: Tell us what happened](${ feedback_link(user_info(), i=feedback_form, github_repo=github_repo_name, github_user=github_user, package_version=package_version_number) } ){:target="_blank"}
+  % endif
+  % if al_enable_incomplete_downloads:
+  * Try to [download your work in progress](${ url_action("al_error_action_download_screen") })
+  % endif
+  % if MAIN_METADATA.get("original_form") and not MAIN_METADATA.get("original_form").strip() == "None":
+  * [Download a blank "${ all_variables(special='metadata').get("title").strip() }"](${ MAIN_METADATA.get("original_form") } )
+  % endif
+  * Visit the [${ AL_ORGANIZATION_TITLE } home page](${ AL_ORGANIZATION_HOMEPAGE }) and try another interview
+  % if al_show_email_to_user_on_errors and get_config("error notification email"):
+  * [Send us an email](mailto:${ get_config("error notification email") }?subject=User%20error%20report%3A%20${ all_variables(special='metadata').get("title") }%20${ user_info().filename }) (Note: you may not get a reply)
+  % endif
```

Comment:
The chance that people actually email us is fairly low, and the auto-emails from errors in interviews have been really useful. Anyway we can just auto-send an email if someone sees this screen, with the same debugging information they would see, or would that require an upstream change

Reply:
Afaict the error emails still happen when you have a custom error action. Just from looking at the code the emails happen before the error action gets triggered

Reply:
Just gave this a test by turning on notifications on the dev server and I can confirm it works even with the custom error action